symphony/bdk/core/datafeed/datafeed_loop_v1.py

```diff
@@ -0,0 +1,48 @@
+from symphony.bdk.core.datafeed.abstract_datafeed_loop import AbstractDatafeedLoop
+from symphony.bdk.core.datafeed.on_disk_datafeed_id_repository import OnDiskDatafeedIdRepository
+from symphony.bdk.gen.exceptions import ApiException
+
+
+class DatafeedLoopV1(AbstractDatafeedLoop):
+
+    def __init__(self, datafeed_api, auth_session, config, repository=None):
+        super().__init__(datafeed_api, auth_session, config)
+        self.datafeed_repository = OnDiskDatafeedIdRepository(config) if repository is None else repository
+        self.datafeed_id = self.retrieve_datafeed()
+        self.started = False
+
+    async def start(self):
+        try:
+            if self.datafeed_id == "":
+                self.datafeed_id = await self.create_datafeed_and_persist()
+            self.started = True
+            while self.started:  # I would like to know why do we use a do while in the java bdk for the read datafeed ?
+                await self.read_datafeed()
+        except ApiException:
```

Comment:
You should filter on HTTP 400 and this should be done inside the while loop.

Reply:
Please tell me if I am wrong but we don't really get a response if the status code is 400, the generated code always throws an exception. **I will look more into this and how to get the status code.**
I tried to handle it like this, but if there is a problem with the read_datafeed endpoint and not the create datafeed endpoint, we end up in an endless loop.
```python
    async def start(self):
        if self.datafeed_id == "":
            self.datafeed_id = await self.create_datafeed_and_persist()
        self.started = True
        while self.started:
            try:
                await self.read_datafeed()
            except ApiException:
                try:
                    self.datafeed_id = await self.create_datafeed_and_persist()
                except ApiException:
                    raise
```

Reply:
You can check this way: 
```
except ApiException as e:
  if e.status == 400:
    ...
```